service/k8sclient/k8sclient.go

```diff
@@ -1662,45 +1677,68 @@ func (c *K8sClient) doAPIRequest(ctx context.Context, method, endpoint string, o
 		}
 	}()
 
-	client := http.Client{
-		Timeout: time.Second * 5,
-		Transport: &http.Transport{
-			MaxIdleConns:    1,
-			IdleConnTimeout: 10 * time.Second,
-		},
-	}
 	req, err := http.NewRequestWithContext(ctx, method, "http://localhost:"+port+"/api"+endpoint, nil)
 	if err != nil {
 		return errors.Wrap(err, "failed to create Kubernetes API request")
 	}
 	var resp *http.Response
 	err = retry.Do(
 		func() error {
-			resp, err = client.Do(req)
-			return err
+			resp, err = c.client.Do(req)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close() //nolint:errcheck
+			return json.NewDecoder(resp.Body).Decode(out)
 		},
 		retry.Context(ctx),
 	)
 	if err != nil {
-		return errors.Wrap(err, "failed to do Kubernetes API request")
+		return errors.Wrap(err, "failed to fetch results from Kubernetes API")
 	}
+	return nil
+}
 
-	defer resp.Body.Close() //nolint:errcheck
-	return json.NewDecoder(resp.Body).Decode(out)
+func (c *K8sClient) getAPIVersionForPSMDBOperator(operators *Operators) string {
+	return fmt.Sprintf(psmdbAPIVersionTemplate, strings.ReplaceAll(operators.Psmdb.Version, ".", "-"))
 }
 
-func (c *K8sClient) InstallXtraDBOperator(ctx context.Context) error {
-	file, err := dbaascontroller.DeployDir.ReadFile("deploy/pxc-operator.yaml")
-	if err != nil {
-		return err
-	}
-	return c.kubeCtl.Apply(ctx, file)
+func (c *K8sClient) getAPIVersionForPXCOperator(operators *Operators) string {
```

Comment:
does it make sense to have this func, if I use it only at one place?

Reply:
i don't really mind having a 1liner method because it makes the code easier to read at the call site rather than reading a long print statement